reports/executive_fullfilment_requests/entrypoint.py

```diff
@@ -0,0 +1,543 @@
+            convert_to_datetime(request.get('updated'))
+            - convert_to_datetime(request.get('created'))
+        ).days
+        _process_vendor_data(report_data, request)
+        _process_product_data(report_data, request)
+        _process_country_data(report_data, request)
+        progress += 1
+        progress_callback(progress, total)
+
+    pdf_reports = {'charts': []}
+
+    chart, total = _generate_pie_chart_from_datat(client, parameters)
+    pdf_reports['charts'].append(
+        {
+            'title': '1. Distribution of requests per type',
+            'description':
+                "Total amount of requests within the period from "
+                f"{parameters['date']['after'].split('T')[0]} to "
+                f"{parameters['date']['before'].split('T')[0]} : "
+                f"<b>{total}<b>.",
+            'images': [chart],
+        },
+    )
+
+    chart, table = _generate_choropleth_map_and_table_from_data(report_data)
+    pdf_reports['charts'].append(
+        {
+            'title': '2. Requests per country',
+            'description':
+                'Following charts represents the request amount per country.'
+                ' The countries that have more than the 20% are near red.',
+            'table': table,
+            'images': [chart],
+        },
+    )
+
+    charts = _generate_vertical_bar_charts_per_type_from_data(report_data)
+    pdf_reports['charts'].append(
+        {
+            'title': '3. Requests per product per type',
+            'description':
+                'Following charts represents the request amount per product.'
+                ' Bar contains the distribution of requests per type.',
+            'images': charts,
+        },
+    )
+
+    charts = _generate_bar_charts_from_data(report_data['vendor'], 'Vendors')
+    pdf_reports['charts'].append(
+        {
+            'title': '4. Averge Request Processing time (per vendor)',
+            'description':
+                'Following charts represents the average processing time of requests per vendor.',
+            'images': charts,
+        },
+    )
+
+    charts = _generate_bar_charts_from_data(report_data['product'], 'Products')
+    pdf_reports['charts'].append(
+        {
+            'title': '5. Averge Request Processing time (per product)',
+            'description':
+                'Following charts represents the average processing time of requests per product.'
+                ' Bar contains the distribution of requests per type.',
+            'images': charts,
+        },
+    )
+
+    account_name, account_id = _get_main_account(client)
+    pdf_reports['range'] = {
+        'start': parameters['date']['after'].split('T')[0],
+        'end': parameters['date']['before'].split('T')[0],
+    }
+    pdf_reports['account_name'] = account_name
+    pdf_reports['account_id'] = account_id
+    pdf_reports['generation_date'] = date.today().strftime('%B %d, %Y')
+    extra_context(pdf_reports)
```

Comment:
There is no need to call extra_context since you return `pdf_reports` from the `generate` function.
This means that you will have the `pdf_reports` data structure available in the template as the `data` variable.